### ISSUE TITLE
DDC-3449: Single scalar Result and Hidden field

### ISSUE THREAD

**Issue by doctrinebot** [doctrine/orm#4257]
Jira issue originally created by user tgallice:

I have try to get a single scalar result but my query contains a HIDDEN field. The result is the `NonUniqueResultException` exception.

**Comment by gitbugr** [comment]
This is still an issue.

**Comment by SenseException** [comment]
This issue needs a better description.

@gitbugr Would you like to add a summary, how to reproduce, the current behavior and the expected behavior to this issue? Having a code example or a PR with a failing test is helpful too.

**Comment by gitbugr** [comment]
@SenseException Sure.

The most basic example of reproducing what is happening above would be: 

```php
$query = $em->createQuery('SELECT COUNT(u.id), u.age AS HIDDEN userAge FROM MyProject\Model\User u HAVING userAge > 20');

$result = $query->getResult();
/**
 *  $result = [
 *    0 => [ 
 *      1 => 1234 
 *    ]
 *  ]
 */

$singleScalarResult = $query->getSingleScalarResult(); // error due to structure of result above
```

The error returned when running `getSingleScalarResult();` would be 
```
Doctrine\ORM\NonUniqueResultException

The query returned a row containing multiple columns. Change the query or use a different result function like getScalarResult().
```

**Comment by SenseException** [comment]
Thank you.

**Comment by gitbugr** [comment]
it's worth noting that a somewhat stable "workaround" for this, that shouldn't break when this gets fixed would be to give the field you're selecting an alias, and returning it like so:

```
$query = $em->createQuery('SELECT COUNT(u.id) AS count, u.age AS HIDDEN userAge FROM MyProject\Model\User u HAVING userAge > 20');

$singleScalarResult = $query->getResult()[0]['count'];
```

**Comment by knezmilos13** [comment]
Same problem here, using doctrine v2.10.1, calling `getSingleScalarResult` when loading `count`, while using another AS HIDDEN attribute, getting the following message:

`The query returned a row containing multiple columns. Change the query or use a different result function like getScalarResult().`

At `\Doctrine\ORM\Internal\Hydration\SingleScalarHydrator.php 51`

**Comment by Mediagone** [comment]
I've submitted a fix to this issue: https://github.com/doctrine/orm/pull/8340

**Comment by greg0ire** [comment]
The fix has been merged.

**Comment by greg0ire** [comment]
Actually, it has been merged in the wrong branch :sweat_smile: 2.7 vs 2.7.x

**Comment by greg0ire** [comment]
See https://github.com/doctrine/orm/pull/9214
